### PR TITLE
Add "Inspect" section to display debug values

### PR DESCRIPTION
**Pull request by afternoon2**
- add `ctx.inspect()` ✅

### CODE REVIEW

File: crates/whiskers/src/runner/mod.rs [hunk 1]
```diff
@@ -157,6 +163,15 @@ impl<A: crate::SketchApp> Runner<'_, A> {
...
     }
+
+    #[must_use]
+    /// Sets the default debug options
+    pub fn with_debug_options(self, options: impl Into<DebugOptions>) -> Self {
+        Self {
+            debug_options: options.into(),
+            ..self
+        }
+    }
```
Comment (abey79) [inline crates/whiskers/src/runner/mod.rs:174]: Let's avoid that.

File: crates/whiskers/src/runner/debug.rs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    }
+}
+
+impl DebugOptions {
+    /// Sets the section's label, as it can serve not only for debugging purposes
+    #[must_use]
+    pub fn label(mut self, value: impl Into<String>) -> Self {
+        self.label = Some(value.into());
+        self
+    }
```
Comment (abey79) [inline crates/whiskers/src/runner/debug.rs:26]: Let's avoid that.

File: crates/whiskers/src/runner/debug.rs [hunk 3]
```diff
@@ -0,0 +1,49 @@
+use std::collections::BTreeMap;
+
+use whiskers_widgets::collapsing_header;
+
+/// Controls the debug section of the runner
+pub struct DebugOptions {
```
Comment (abey79) [inline crates/whiskers/src/runner/debug.rs:6]: Should be renamed `InspectVariables`

File: crates/whiskers/src/runner/mod.rs [hunk 4]
```diff
@@ -17,6 +18,7 @@ use std::sync::Arc;
 use crate::Sketch;
 pub use animation::AnimationOptions;
 use convert_case::Casing;
+pub use debug::DebugOptions;
```
Comment (abey79) [inline crates/whiskers/src/runner/mod.rs:21]: Not `pub`

**Review by abey79 — COMMENTED** [review]
This is a great feature!

I have yet to give it a test, but I already have a bit of a gripe with name configurability: I'd much rather do without. I do agree that "Debug" may sound a bit too low-level/"scary" for something that might show up in release/final sketches. Instead, we could name the panel "Inspect" and the function `ctx.inspect()`.

**Comment by afternoon2** [comment]
As a user, I'd still opt for a rename possibility (inspect sounds less alarming, but still in a "checking if everything's fine" manner rather than presenting program's valid features). And if not via the runner's method, then maybe with `ctx.inspect_label(label: &str)`?

**Comment by abey79** [comment]
> As a user, I'd still opt for a rename possibility (inspect sounds less alarming, but still in a "checking if everything's fine" manner rather than presenting program's valid features). And if not via the runner's method, then maybe with `ctx.inspect_label(label: &str)`?

This really feels like API clutter, regardless of where it lives. What's the use case here? What problem would that solve?

File: crates/whiskers/src/runner/mod.rs (hunk 4 above)
Comment (afternoon2) [reply crates/whiskers/src/runner/mod.rs]: here it's `pub` because it's used by the context. but I removed it from `lib.rs` and `prelude.rs`. is it ok?

**Comment by afternoon2** [comment]
just a customization option, but I won't fight for it

**Comment by afternoon2** [comment]
ah, forgot to add a label for the PR, sorry 🙏🏻

**Comment by abey79** [comment]
> ah, forgot to add a label for the PR, sorry 🙏🏻

No prob, I'm not even sure they are settable by external contributors 🤷🏻

**Comment by abey79** [comment]
Sorry for the humongous delay! 🙏🏻

**Pull request merged**